src/amount.rs

```diff
@@ -168,19 +211,46 @@ impl Amount {
         ]
         .iter()
         .cloned()
```

Comment:
Can you use `into_iter` here?
```suggestion
        .into_iter()
```

Reply:
yep!

Reply:
whoops, I mean no!    I tried that before also, and get:

```
error[E0277]: a value of type `BTreeMap<amount::Amount, &str>` cannot be built from an iterator over elements of type `&(amount::Amount, &str)`
   --> src/amount.rs:983:10
    |
983 |         .collect()
    |          ^^^^^^^ value of type `BTreeMap<amount::Amount, &str>` cannot be built from `std::iter::Iterator<Item=&(amount::Amount, &str)>`
    |
    = help: the trait `std::iter::FromIterator<&(amount::Amount, &str)>` is not implemented for `BTreeMap<amount::Amount, &str>`
```